app/assets/javascripts/pretty-text/engines/discourse-markdown/quotes.js

```diff
@@ -59,9 +68,9 @@ const rule = {
     }
 
     if (options.formatUsername) {
-      displayName = options.formatUsername(username);
+      displayName ||= options.formatUsername(username);
```

Comment:
Maybe we should avoid this fancy syntax for now, it risks breaking things for oldish browsers like Safari on iOS 12/13.